### ISSUE TITLE
[Tracking] [scaffolding-chef-infra] Rubygems environment causes issues with building some packages

### ISSUE THREAD

**Issue by echohack** [chef/effortless#54]
When I run the build I get an error that is documented that requires you to downgrade the chef ruby gem as a workaround (Reason: (ArgumentError) "\x80\x00\x00\x00d\x9E\xD3\xD9" is not an octal string)
```
* https://github.com/rubygems/rubygems/issues/2213 - the current remediation recipe code fails to resolve dependencies due to this rubygems bug. You will see an error during `kitchen converge` like `Reason: (ArgumentError) "\x80\x00\x00\x00d\x9E\xD3\xD9" is not an octal string`. The workaround is to downgrade rubygems:  chef exec gem update --system 2.7.5```

## Software Version
scaffolding-chef-infra 0.8.0 or lesser

## Replication Case
Build a package using https://github.com/chef/cis-rhel as a cookbook in the policyfile and runlist.

## Workaround

You can simply use an older version of the chef-dk to downgrade your rubygems environment. 

```
pkg_name=cis-rhel
pkg_origin=echohack
pkg_version="0.1.0"
pkg_maintainer="The Habitat Maintainers <humans@habitat.sh>"
pkg_license=("Apache-2.0")
pkg_upstream_url="http://chef.io"
pkg_scaffolding="chef/scaffolding-chef-infra"
scaffold_policy_name="cis-rhel"
scaffold_policyfile_path="$PLAN_CONTEXT"
scaffold_chef_dk="chef/chef-dk/3.0.38"
```

## Possible Solution
`chef/chef-dk` will need to upgrade it's rubygems environment to version `3.0.5`, which should fix this issue.

This is marked as a tracking issue as the issue may resolve itself without updating code in this repository.

**Comment by echohack** [comment]
Looks like Chef-DK builds have been disabled for some time. We'll need to fix this in order for any bug fixes to land.

https://github.com/chef/chef-dk/blob/master/.expeditor/config.yml#L88

**Comment by TheLunaticScripter** [comment]
As stated by the wise @robbkidd 
```
The two factors I can think of:
- the version resolved of one of the cookbooks being retrieved—either a direct dependency or a transitive dependency—has a large user or group ID recorded in its tarball (the cookbook version was probably published from a Windows host or a Window file share on a Linux host in an Active Directory environment)
- the build environment trying to retrieve this cookbook includes a RubyGems install from a specific range of versions that doesn't know how to handle large uid/gid values recorded in a tarball (true for many builds of DK)
```

**Comment by TheLunaticScripter** [comment]
The Chef DK is released for Linux at 4.5.0 so this should no longer be a Chef DK problem as we are publishing new versions I'm going to close this issue but should it come up again I think looking at the reasons above should be the first step.
